MapboxDirections/MBSpeedLimit.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  MBSpeedLimit.swift
+//  MapboxDirections
+//
+//  Created by Avi Cieplinski on 4/15/19.
+//  Copyright © 2019 Mapbox. All rights reserved.
+//
+
+import Foundation
+
+@objc(MBSpeedLimit)
+public class SpeedLimit: NSObject {
+
+    @objc public static let invalid = Double(-1)
```

Comment:
I think we should offer a way for the developer to easily check whether the speed limit as a whole is invalid (unset), not just its value. We could define:

```swift
@objc public static let invalid = SpeedLimit(value: -1, unit: .kilometersPerHour)
```

plus an `isEqual(_:)` implementation, so that the developer could check:

```swift
if maximumSpeed == .invalid { … }
```

/ref https://github.com/mapbox/MapboxDirections.swift/pull/250#discussion_r176279345